### PR TITLE
Drawer - Performance

### PR DESCRIPTION
**Pull request by Inbal-Tish**

### CODE REVIEW

File: demo/src/screens/listScreens/ConversationListScreen.js [hunk 1]
```diff
@@ -1,113 +1,197 @@
-import React, {Component} from 'react';
```
Comment (ethanshar) [inline demo/src/screens/listScreens/ConversationListScreen.js]: The performance is much better, but I think I notice a bug. 
open the drawer of one of the items, then close its drawer, then try to open the same item's drawer again. it won't open, it keeps getting closed.

File: demo/src/screens/listScreens/ConversationListScreen.js [hunk 2]
```diff
@@ -1,113 +1,197 @@
...
+        data={this.state.items}
+        renderItem={this.renderItem}
         keyExtractor={this.keyExtractor}
+        onEndReached={this.onEndReached}
       />
     );
   }
 }
 
+export default gestureHandlerRootHOC(ConversationListScreen);
```
Comment (ethanshar) [inline demo/src/screens/listScreens/ConversationListScreen.js]: move to the end of the file

File: src/interactableComponents/drawer/index.js [hunk 3]
```diff
@@ -3,11 +3,15 @@ import PropTypes from 'prop-types';
...
 import Interactable from 'react-native-interactable';
-import {BaseComponent} from '../../commons';
+// import {BaseComponent} from '../../commons';
+import baseComponent from './../../commons/baseComponent';
 import {screenWidth} from '../../helpers/Constants';
 import {Colors, Typography} from '../../style';
 import NewDrawer from './newDrawer';
 
+
+const BaseComponent = baseComponent(true);
```
Comment (ethanshar) [inline src/interactableComponents/drawer/index.js]: you don't need to do that (: 
you can just import `PureBaseComponent` from ` '../../commons'`

File: src/interactableComponents/drawer/newDrawer.js [hunk 4]
```diff
@@ -20,23 +20,29 @@ const ITEM_PROP_TYPES = {
   onPress: PropTypes.func,
   keepOpen: PropTypes.bool,
   style: ViewPropTypes.style,
-  testID: PropTypes.string,
+  testID: PropTypes.string
 };
 
-export default class NewDrawer extends BaseComponent {
+// TODO: change to pure component after deprecation of old drawer
```
Comment (ethanshar) [inline src/interactableComponents/drawer/newDrawer.js]: Why not now? Does it break anything?

File: demo/src/screens/listScreens/ConversationListScreen.js (hunk 1 above)
Comment (Inbal-Tish) [reply demo/src/screens/listScreens/ConversationListScreen.js]: Fixed. Thanks!

File: demo/src/screens/listScreens/ConversationListScreen.js (hunk 2 above)
Comment (Inbal-Tish) [reply demo/src/screens/listScreens/ConversationListScreen.js]: Done.

File: src/interactableComponents/drawer/index.js (hunk 3 above)
Comment (Inbal-Tish) [reply src/interactableComponents/drawer/index.js]: Ok. Done.

File: src/interactableComponents/drawer/newDrawer.js (hunk 4 above)
Comment (Inbal-Tish) [reply src/interactableComponents/drawer/newDrawer.js]: Done.

**Pull request merged**